home/serializers.py

```diff
@@ -359,6 +415,106 @@ def body_field_representation(page: Any, request: Any) -> Any:
+    ):
+        if page.sms_body != []:
+            try:
+                return OrderedDict(
+                    [
+                        ("message", message + 1),
+                        (
+                            "next_message",
+                            has_next_message(message, page, "sms"),
+                        ),
+                        (
+                            "previous_message",
+                            has_previous_message(message, page, "sms"),
+                        ),
+                        ("total_messages", len(page.sms_body._raw_data)),
+                        ("text", page.sms_body._raw_data[message]["value"]),
+                    ]
+                )
+            except IndexError:
+                raise ValidationError("The requested message does not exist")
+    elif "ussd" in request.GET and (
+        page.enable_ussd is True
+        or ("qa" in request.GET and request.GET["qa"] == "True")
+    ):
+        if page.ussd_body != []:
+            try:
+                return OrderedDict(
+                    [
+                        ("message", message + 1),
+                        (
+                            "next_message",
+                            has_next_message(message, page, "ussd"),
+                        ),
+                        (
+                            "previous_message",
+                            has_previous_message(message, page, "ussd"),
+                        ),
+                        ("total_messages", len(page.ussd_body._raw_data)),
+                        ("text", page.ussd_body._raw_data[message]["value"]),
+                    ]
+                )
+            except IndexError:
+                raise ValidationError("The requested message does not exist")
+    elif "messenger" in request.GET and (
+        page.enable_messenger is True
+        or ("qa" in request.GET and request.GET["qa"] == "True")
+    ):
+        if page.messenger_body != []:
+            try:
+                return OrderedDict(
+                    [
+                        ("message", message + 1),
+                        (
+                            "next_message",
+                            has_next_message(message, page, "messenger"),
+                        ),
+                        (
+                            "previous_message",
+                            has_previous_message(message, page, "messenger"),
+                        ),
+                        ("total_messages", len(page.messenger_body._raw_data)),
+                        ("text", page.messenger_body._raw_data[message]["value"]),
+                    ]
+                )
+            except IndexError:
+                raise ValidationError("The requested message does not exist")
+    elif "viber" in request.GET and (
+        page.enable_viber is True
+        or ("qa" in request.GET and request.GET["qa"] == "True")
```

Comment:
I don't think that this is correct logic, if `enable_viber` is `False` in the latest revision, then we should not show it, but this logic will. I think this whole section needs a refactor, but if we're doing that in the next ticket, then I'm okay with this going through in this one.